internal/controller/promotions/promotions.go

```diff
@@ -422,6 +421,13 @@ func (r *reconciler) promote(
 	if err != nil {
 		return nil, err
 	}
+	// record the freight reference to the promotion's status
+	err = kubeclient.PatchStatus(ctx, r.kargoClient, &promo, func(status *kargoapi.PromotionStatus) {
+		status.PromotedFreight = &targetFreightRef
+	})
+	if err != nil {
+		return nil, err
+	}
 
 	newStatus, nextFreight, err := r.promoMechanisms.Promote(ctx, stage, &promo, targetFreightRef)
```

Comment:
If I recall correctly, `Promote` may actually modify the `FreightRef` with more accurate information based on the outcome from the promotion mechanisms. Given this, I think only patching the Status _before_ promoting doesn't suffice.

Reply:
Can you please kindly help to point out where it was mentioned?

Reply:
An example of where this happens is during Git promotion, where a `HealthCheckCommit` gets added to the `FreightRef`: https://github.com/akuity/kargo/blob/38ae6e6fedcfcc8da320e912347d27fae94f5d02/internal/controller/promotion/git.go#L209-L211

Reply:
Spot on. The health check commit is the main thing that may _change_ in the course of promotion.

To be perfectly honest, it's not going to matter to any health check logic if we don't update it the `PromotionStatus`, but it may cause confusion among human users if they notice the discrepancy between the FreightReference held by the Stage's CurrentFreight and History fields and this one.

Reply:
i.e. So we _should_ make sure it gets set here.